0000-interledger-protocol-4/0000-interledger-protocol-4.md

```diff
@@ -19,44 +19,58 @@ Interledger is a protocol for sending packets of money across different payment
-- **Bandwidth** - Connectors may limit the total value of ILP packets a customer or peer can send or receive within a given time period. This may be limited by the minimum balance on that party's account, which can be refilled by settling outstanding debts through a ledger or payment channel. Connectors may also limit each account's bandwidth to prevent one customer or peer from using or tying up all of the connector's liquidity.
-- **Condition and Fulfillment** - ILP packets are secured with conditions so that they cannot be lost or stolen by intermediary connectors. ILP Prepare packets are sent with a condition or SHA-256 hash in them. When the receiver receives the packet, they present the fulfillment, or valid preimage of the hash, to execute the transfer.
-- **Connector** - Intermediary that maintains settlement accounts on one or more ledgers or payment systems. Connectors forward ILP packets. They may generate revenue from spreads on currency conversion, through subscription fees, or other means.
+- **Account** - Two peers establish an account with one another where they track the current obligations they hold with one another; i.e. the balance on the account. The account is denominated in a single asset. The peers agree on the precision and scale they will use for the amounts of that asset when they express amounts in messages between one another. Accounts may be pre-funded (in which case the customer or peer trusts the connector for the value of the account balance) or post-funded (in which case the connector trusts the customer or peer for the amount the account is allowed to go negative). Connectors may refuse to process further ILP packets if an account balance goes too low. Note that the mentions of "accounts" in this document refers to those that Interledger nodes _hold with one another_, rather than the "accounts" they may hold on underlying ledgers because the latter are not directly part of the ILPv4 packet flow. 
+- **Bandwidth** - Connectors may limit the total value of ILP packets an account can have in-flight (prepared but not yet fulfilled or rejected) at the same time. The limit may either be based on how much the connector trusts the account-holder, represented by the minimum balance limit on the account, or it may be used to prevent one account from tying up all of the connector's bandwidth with its peers.
+- **Condition and Fulfillment** - ILP packets are secured with conditions so that they cannot be lost or stolen by intermediary connectors. ILP Prepare packets are sent with a condition or SHA-256 hash in them. When the receiver receives the packet, they present the fulfillment, or valid preimage of the hash, to execute the transfer. Unlike in earlier versions of ILP, the conditions and fulfillments are NOT assumed to be enforced by the underlying ledger but they help peers reconcile their account balances. Two peers should only consider an ILP packet executed (and therefore affect their account balances) if 
```

Comment:
> ILP Prepare packets are sent with a condition or SHA-256 hash in them. When the receiver receives the packet, they present the fulfillment, or valid preimage of the hash, to execute the transfer.

Is this intended to leave open the possibility that a condition might _not_ be a SHA-256 hash? That's the implication I read from this phrasing. If that's not intended, I recommend the following rephrase:

> ILP Prepare packets contain a _condition_ in the form of a SHA-256 hash. When the receiver receives the packet, they present the _fulfillment_, in the form of a valid preimage of the condition hash, to execute the transfer.